Use new CoreDisTools package

echesakov opened the pull request:
After the change R2RDump and GCStress will use a newer version of coredistools produced from https://github.com/dotnet/jitutils

echesakov committed:
Add eng/coredistools.targets and define MicrosoftNETCoreCoreDisToolsVersion in eng/Versions.props

echesakov committed:
Copy an appropriate coredistools native library to output directory in src/coreclr/src/tools/r2rdump/R2RDump.csproj

echesakov committed:
Copy an appropriate coredistools native library to Core_Root in src/tests/Common/Directory.Build.targets

echesakov committed:
Remove old mechanism for restoring stress dependencies in src/tests/build.cmd and src/tests/build.sh

echesakov committed:
Stop using skipstressdependencies in *.yml

echesakovMSFT commented:
I believe this should be ready for review - I ran gcstress pipeline and it shows the same behaviour as last Sunday run (https://dev.azure.com/dnceng/public/_build/results?buildId=879006) - it took the same time to finish and produces the same tests failures *(that I believe should now be fixed)*.

PTAL @dotnet/jit-contrib @dotnet/crossgen-contrib @dotnet/runtime-infrastructure I would like to have three sign-offs from a JIT team member (GCStress), an AOT team member (R2RDump) and someone who knows about the MSBuild infrastructure.

nattress reviewed (approved):
The R2RDump changes LGTM - thanks for centralizing the management of that package!

trylek commented on src/coreclr/src/tools/r2rdump/R2RDump.csproj:
```diff
@@ -15,15 +15,16 @@
       <Version>$(SystemCommandLineVersion)</Version>
     </PackageReference>
     <ProjectReference Include="..\aot\ILCompiler.Reflection.ReadyToRun\ILCompiler.Reflection.ReadyToRun.csproj" />
+    <Content Include="$(CoreDisToolsLibrary)" Condition="Exists('$(CoreDisToolsLibrary)')">
```
What's the purpose of the <code>Exists</code> check, isn't that rather hiding a build inconsistency? I was under the impression that thanks to your work we now have <code>coredistools</code> available on all supported platforms.

echesakovMSFT replied:
There are no coredistools library for linux-musl and some unsupported platforms (such as linux-x86). Since R2RDump is a .NET Core app, in theory, it can be built targeting these platforms. The check is to prevent the build from failure (but, obviously, the disassembler would not work).

trylek replied:
I see, thanks for clarifying!

trylek reviewed (commented):
Thanks a lot for the wonderful test build script cleanup on top of this super-useful feature!

janvorli reviewed (approved):
LGTM, thank you a lot for making this change and the thorough cleanup of the build scripts.

janvorli commented:
I can see the R2RDumpTests is failing on all platforms with timeout in the CI tests.

trylek commented:
R2RDumpTests is a recently introduced test that I only earlier this week fixed to pass in Crossgen2 runs; after the fix I noticed yesterday that it still fails in the runtime-coreclr crossgen2 composite gcstress pipeline making me believe that the failure is only specific to GC stress testing. I'll go ahead and mark the test as GC stress-incompatible because its intent is not to stress the runtime - in fact the test is kind of "lengthy" in terms of CoreCLR unit tests as it disassembles the entire SPC assembly; I can easily imagine that the slowdown by an order of magnitude or more incurred by the GC stress-instrumented code can easily push it past the timeout. The test also runs in all CoreCLR test legs of the "runtime" pipeline that seem to be passing just fine.

ViktorHofer commented on src/tests/Common/Directory.Build.targets:
```diff
@@ -10,6 +10,14 @@
+    <GCStressDependsOnCoreDisTools Condition="'$(TargetOS)' == 'Linux' And '$(TargetArchitecture)' == 'x64'">true</GCStressDependsOnCoreDisTools>
+  </PropertyGroup>
+
+  <Import Project="$(RepoRoot)eng/coredistools.targets" Condition="$(GCStressDependsOnCoreDisTools)" />
```
```suggestion
  <Import Project="$(RepositoryEngineeringDir)coredistools.targets" Condition="$(GCStressDependsOnCoreDisTools)" />
```

ViktorHofer commented on eng/coredistools.targets:
```diff
@@ -0,0 +1,15 @@
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. -->
+<Project>
+  <ItemGroup Condition="'$(DotNetBuildFromSource)' != 'true'">
```
Should the whole file import instead be conditioned as the properties below won't work without the PackageReference.

echesakovMSFT replied:
I can move this condition to `Import` element in R2RDump.csproj. 

Would I need the same condition for `src/tests/Common/Directory.Build.targets` ? As far as I understand, we don't build tests during source build, so the condition will always be true.

ViktorHofer replied:
Yeah please add it to both imports. You are right that we don't build these currently in source build we might at some point.

echesakovMSFT replied:
Thanks @ViktorHofer - I addressed both of your suggestions.

sandreenko reviewed (approved):
GCStress changes LGTM

trylek commented:
I've merged in the change disabling R2RDumpTests in GC stress mode; this should make the "runtime-coreclr gcstress0x3-gcstress0xc" legs green.

echesakov committed:
Address Viktor's feedback:
  1. Use $(RepositoryEngineeringDir)
  2. Remove condition on ItemGroup in coredistools.targets and
  3. Instead use the condition on Import elements in R2RDump.csproj and tests *.targets

echesakov committed:
Merge branch 'master' into Use-New-CoreDisTools

echesakovMSFT commented:
> I've merged in the change disabling R2RDumpTests in GC stress mode; this should make the "runtime-coreclr gcstress0x3-gcstress0xc" legs green.

Thank you Tomas, I merged master on top of the PR changes and will re-start gcstress pipelines

ViktorHofer commented on eng/coredistools.targets:
```diff
@@ -0,0 +1,15 @@
+  </ItemGroup>
+
+  <PropertyGroup>
+    <CoreDisToolsLibrary Condition="'$(TargetOS)' == 'Windows'">$(PkgMicrosoft_NETCore_CoreDisTools)\runtimes\win-$(TargetArchitecture)\native\coredistools.dll</CoreDisToolsLibrary>
```
@Anipik `TargetOS` for Windows should be lowercase `windows`, right?

safern replied:
That is correct.

echesakovMSFT replied:
Let me follow up on this

ViktorHofer replied:
Thanks. I saw this in another place in your PR as well. It isn't causing a problem right now as msbuild is case insensitive but it should be cleaned up.

echesakovMSFT replied:
Should be fixed in https://github.com/dotnet/runtime/pull/44845

AntonLapounov replied:
What was the reason to have `Linux` and `OSX` uppercased but not `Windows`?

ViktorHofer reviewed (commented):
LGTM. Thanks
